jwst/pipeline/calwebb_spec2.py

```diff
@@ -290,29 +288,42 @@ def process_exposure_product(
         if exp_type in ['NRS_FIXEDSLIT', 'NRS_MSASPEC', 'MIR_LRS-FIXEDSLIT'] \
            and not isinstance(calibrated, datamodels.CubeModel):
 
-            # Call the resample_spec step for 2D slit data
+            # Call pixel replace, followed by resample_spec  for 2D slit data
             resampled = calibrated.copy()
+            self.log.info("calling pixel replace NRS FS, MSA, LRS FIXED SLIT")
+            resampled = self.pixel_replace(resampled)
+            self.log.info('pixel replace status %s',resampled.meta.cal_step.pixel_replace)
             resampled = self.resample_spec(resampled)
 
+
         elif is_nrs_slit_linelamp(calibrated):
 
-            # Call resample_spec for NRS 2D line lamp slit data
+            # Call pixel_replace followed by resample_spec for NRS 2D line lamp slit data
             resampled = calibrated.copy()
+            resampled = self.pixel_replace(resampled)
             resampled = self.resample_spec(resampled)
 
         elif (exp_type in ['MIR_MRS', 'NRS_IFU']) or is_nrs_ifu_linelamp(calibrated):
 
-            # Call the cube_build step for IFU data;
-            # always create a single cube containing multiple
+            # First call pixel_replace then call cube_build step for IFU data.
+            # For cube_build always create a single cube containing multiple
             # wavelength bands
 
             resampled = calibrated.copy()
+            resampled = self.pixel_replace(resampled)
             resampled = self.cube_build(resampled)
             if not self.cube_build.skip:
                 self.save_model(resampled[0], suffix='s3d')
+        elif exp_type in ['MIR_LRS-SLITLESS']:
+            resampled = calibrated.copy()
+            resampled = self.pixel_replace(resampled)
         else:
-            resampled = calibrated
-
+            # pixel replacement for all other modes
+            # will be run if set in parameter ref file or by user
+            resampled = calibrated.copy()
+            self.log.info("calling pixel replace on other modes")
```

Comment:
Again, don't need these log messages before and after the step. The step will create it's own messages.